controllers/cluster_controller.go

```diff
@@ -417,6 +434,10 @@ func (c clusterDescendants) filterOwnedDescendants(cluster *clusterv1.Cluster) (
 		&c.workerMachines,
 		&c.controlPlaneMachines,
 	}
+	if feature.Gates.Enabled(feature.MachinePool) {
+		lists = append(lists, &c.machinePools)
```

Comment:
Just realized that we should probably not be appending the machinePools to the end of the list, but rather pre-pending them (or at least inserting them prior to `controlPlaneMachines`.

We can handle that as a followup, though.

Reply:
Might be worthwhile to do it within this PR, or wait for the next release

Reply:
Updating